perfkitbenchmarker/disk.py

```diff
@@ -45,6 +93,67 @@ def __init__(cls, name, bases, dct):
     super(AutoRegisterDiskSpecMeta, cls).__init__(name, bases, dct)
 
 
+def WarnAndTranslateDiskTypes(name, cloud):
+  """Translate old disk types to new disk types, printing warnings if needed.
+
+  Args:
+    name: a string specifying a disk type, either new or old.
+    cloud: the cloud we're running on.
+
+  Returns:
+    The new-style disk type name (i.e. the provider's name for the type).
+
+  Raises:
+    ValueError, if name is not a disk type name.
+  """
+
+  if cloud in DISK_TYPE_MAPS:
+    disk_type_map = DISK_TYPE_MAPS[cloud]
+    if name in disk_type_map and disk_type_map[name] != name:
+      new_name = disk_type_map[name]
+      logging.warning('Disk type name %s is deprecated and will be removed. '
+                      'Translating to %s for now.', name, new_name)
+      return new_name
+    else:
+      return name
+  else:
+    logging.info('No legacy->new disk type map for provider %s', cloud)
+    # The provider has not been updated to use new-style names. We
+    # need to keep benchmarks working, so we pass through the name.
+    return name
+
+
+def WarnAndCopyFlag(old_name, new_name):
+  """Copy a value from an old flag to a new one, warning the user.
+  """
+
+  if FLAGS[old_name].present:
+    logging.warning('Flag --%s is deprecated and will be removed. Please '
+                    'switch to --%s.', old_name, new_name)
+    if not FLAGS[new_name].present:
+      FLAGS[new_name].value = FLAGS[old_name].value
+
+      # Mark the new flag as present so we'll print it out in our list
+      # of flag values.
+      FLAGS[new_name].present = True
+    else:
+      logging.warning('Ignoring legacy flag %s because new flag %s is present.',
+                      old_name, new_name)
+  # We keep the old flag around so that providers that haven't been
+  # updated yet will continue to work.
+
+
+def WarnAndTranslateDiskFlags():
+  """Translate old disk-related flags to new disk-related flags.
+  """
+
+  WarnAndCopyFlag('scratch_disk_type', 'data_disk_type')
```

Comment:
I think it is cleaner to put old flag, new flag into a constant dict. And loop through the dict to copy old flags into new flags.